conda/recipes/cudf/meta.yaml

```diff
@@ -82,7 +83,8 @@ requirements:
     - pandas >=2.0,<2.2.3dev0
     - cupy >=12.0.0
     - numba >=0.57
-    - {{ pin_compatible('numpy', max_pin='x') }}
+    # TODO: Update `numpy` in `host` when dropping `<2.0a0`
+    - numpy >=1.23,<2.0a0
```

Comment:
While this seems duplicative to what `numpy`'s `run_exports` does, this same pinning is used throughout RAPIDS. Also it gives a foothold for RDFG changes to be made to move to NumPy 2. When that does happen, we have a comment above to remind us to make other needed changes in this recipe